packages/x-tree-view/src/internals/plugins/useTreeViewKeyboardNavigation/useTreeViewKeyboardNavigation.ts

```diff
@@ -137,159 +102,220 @@ export const useTreeViewKeyboardNavigation: TreeViewPlugin<
       index = findNextFirstChar(firstChars, 0, lowercaseChar);
     }
 
-    // If match was found...
+    // If a match was found...
     if (index > -1) {
-      instance.focusNode(event, firstCharIds[index]);
+      return firstCharIds[index];
     }
-  };
 
-  const selectNextNode = (event: React.KeyboardEvent<HTMLUListElement>, id: string) => {
-    if (!instance.isNodeDisabled(getNextNode(instance, id))) {
-      instance.selectRange(
-        event,
-        {
-          end: getNextNode(instance, id),
-          current: id,
-        },
-        true,
-      );
-    }
+    return null;
   };
 
-  const selectPreviousNode = (event: React.KeyboardEvent<HTMLUListElement>, nodeId: string) => {
-    if (!instance.isNodeDisabled(getPreviousNode(instance, nodeId))) {
-      instance.selectRange(
-        event,
-        {
-          end: getPreviousNode(instance, nodeId)!,
-          current: nodeId,
-        },
-        true,
-      );
-    }
-  };
+  const canToggleNodeSelection = (nodeId: string) =>
+    !params.disableSelection && !instance.isNodeDisabled(nodeId);
+
+  const canToggleNodeExpansion = (nodeId: string) =>
+    !instance.isNodeDisabled(nodeId) && instance.isNodeExpandable(nodeId);
 
+  // ARIA specification: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/#keyboardinteraction
   const createHandleKeyDown =
-    (otherHandlers: EventHandlers) => (event: React.KeyboardEvent<HTMLUListElement>) => {
+    (otherHandlers: EventHandlers) =>
+    (event: React.KeyboardEvent<HTMLUListElement> & MuiCancellableEvent) => {
       otherHandlers.onKeyDown?.(event);
 
-      let flag = false;
-      const key = event.key;
+      if (event.defaultMuiPrevented) {
```

Comment:
Do you think that this feature will be transparent enough for the users? 🤔 
Have you considered checking `event.defaultPrevented` instead?
It seems that in cases, where this behavior is supported, we do explicitly [document it](https://mui.com/material-ui/react-autocomplete/#events). 🤔 
If we go with `defaultMuiPrevented`, do we agree that it should be explicitly documented?

Reply:
`defaultMuiPrevented` is a pattern that a several recent core components uses and the grid as well.
I think the reasoning behind it is to be able to distinguish browser behavior and MUI behavior. To let people say "hey I want the browser behavior, but not the MUI one".

With that being said, I can't find any doc about it except in this tech file: https://github.com/mui/material-ui/blob/master/packages/mui-base/CONTRIBUTING.md
Once the pattern is widely used in the MUI codebase, it should probably have a dedicated doc section somewhere.
And the Tree View `onKeyDown` event is probably common enough to justify a doc examples as well.

Reply:
Maybe @michaldudak has more context here

Reply:
AFAIK Material UI uses this pattern in several places, and I adopted it in Base UI as well (it was necessary to have an option to prevent just the library code from executing).
I agree about the docs lacking in this area. I'm making a note to update them.

Reply:
@flaviendelangle If you think that not providing an explicit documentation section about this possibility in the current state of all products documentation is fine, I'm not blocking the PR merging, but IMHO, such a section wouldn't hurt. 😉

Reply:
I was wondering if this shouldn't be addressed at the API page level.
To have the `onKeyDown` prop listed [here](https://next.mui.com/x/api/tree-view/rich-tree-view/) with a description that sends to a MUI-wide page explaining how to use `defaultMuiPrevented`.

If we want to do it on the product pages, we could create a new "Interaction" pages on both `SimpleTreeView` and `RichTreeView` and add a section "Customize the keyboard behaviors".
Or we wait for the "Accessibility" page and we put it there since it will be the page that explains the keyboard behavior in depth.

What do you think?

Reply:
Good points. 👍 
The first suggestion sounds the best, but AFAIK, it's more or less a future solution.
I'm not against just waiting a bit until there is a common page that we can link to. 🤔 

Adding this info to the `Accessibility` page is also an option, but I'd argue that it is not the best place for such content. 🤔 
We could link to the above-mentioned page there. 💡

I'm not the biggest fan of the `Interaction` page suggestion if it would only include this information, but IMHO it's a better alternative than the `Accessibility` page. 🤔

Reply:
We could re-purpose the `Accessibility` page to be more a `Keyboard usage` page (this name is terrible though).

Tbh I did not add this doc section because I did not find a good place to put it :laughing:

Reply:
I'll merge without a doc, we can discuss later what we want to do :+1: